internal/config/validation.go

```diff
@@ -17,6 +17,19 @@ import (
 	"github.com/go-playground/validator/v10"
 )
 
+func validateImage(fl validator.FieldLevel) bool {
+	image := fl.Field().String()
+	switch {
+	case image == "":
+		return false
+	case image == "..":
+		return false
+	case strings.Contains(image, "/"):
+		return false
+	}
+	return true
+}
+
```

Comment:
This version field is used in multiple places. It is used as part of a URL, it is used as part of some filenames. This is the best way I can think of to ensure  that it's safe. I could also add additional checks to every single consumer of this field.

Reply:
In any case, having strict config validation is preferable in my opinion as it catches user errors early

Reply:
I was more thinking of some kind of sanitation like [this](https://dzx.cz/2021/04/02/go_path_traversal/).

 Then theoretically that check could be removed as the actual implementation should be more resistent (if that actually works, haven’t tested it).

Reply:
Ah, the point mainly was that based on validation name & code that go-validator shouldn't be security code to prevent path traversal.

If the underlying path code cleans up and avoids this then all is fine, but you can also keep the input validation on top if you'd like. It just shouldn't be a security check.